packages/arb-node-core/ethbridge/validator.go

```diff
@@ -93,6 +104,27 @@ func (v *ValidatorWallet) executeTransaction(ctx context.Context, tx *types.Tran
 	})
 }
 
+func (v *ValidatorWallet) createWalletIfNeeded(ctx context.Context) error {
+	if v.address == nil {
+		addr, err := CreateValidatorWallet(ctx, v.walletFactoryAddr, v.rollupFromBlock, v.auth, v.client)
+		if err != nil {
+			return err
+		}
+		v.address = &addr
+		if v.onWalletCreated != nil {
+			v.onWalletCreated(addr)
+		}
+	}
+	if v.con == nil {
```

Comment:
I think we might as well do this unconditionally if we're setting the address

Reply:
Fixed as part of the flattening